src/coreclr/src/jit/optcse.cpp

```diff
@@ -459,14 +476,36 @@ unsigned Compiler::optValnumCSE_Index(GenTree* tree, Statement* stmt)
         //
         assert(vnLibNorm == vnStore->VNNormalValue(vnOp2Lib));
     }
-    else // Not a GT_COMMA
+    else if (enableSharedConstCSE && tree->IsIntegralConst())
+    {
+        assert(vnStore->IsVNConstant(vnLibNorm));
+        key = vnStore->CoercedConstantValue<size_t>(vnLibNorm);
+
+        // We don't shared small offset constants when we require a reloc
+        if (!tree->AsIntConCommon()->ImmedValNeedsReloc(this))
+        {
+            // Make constants that have the same upper bits use the same key
+
+            // Shift the key right by CSE_CONST_SHARED_LOW_BITS bits and set the upper bits to zero
```

Comment:
Where do we do `set the upper bits to zero` from the comment?

Reply:
The arithmetic right shift zeros out the upper bits

Reply:
Thanks, missed that it is unsigned.